### PR TITLE
Go1.20 bump & boilerplate update

### PR DESCRIPTION
**Pull request by a7vicky**
### What type of PR is this?
_(refactor)_

### What this PR does / why we need it?

- Update boilerplate to the latest version
- Update Golang to 1.20 

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_[OSD-20177](https://issues.redhat.com/browse/OSD-20177)

### CODE REVIEW

**Comment by a7vicky** [comment]
/hold
For testing

File: go.mod [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.62.0
 	github.com/prometheus/client_golang v1.15.1
 	github.com/sykesm/zap-logfmt v0.0.4
 	go.uber.org/zap v1.24.0
-	k8s.io/api v0.28.2
-	k8s.io/apimachinery v0.28.2
-	k8s.io/client-go v0.28.2
+	k8s.io/api v0.29.2
+	k8s.io/apimachinery v0.29.2
+	k8s.io/client-go v0.29.1
```
Comment (ravitri) [inline go.mod:19]: Can you align it to 0.28.2 as per https://github.com/openshift/api/blob/release-4.15/go.mod#L8 ? Let's tie the version to 4.15 OCP release branch

**Comment by ravitri** [comment]
/hold

Adding hold to fix the go mod update for 4.15 release

File: go.mod (hunk 1 above)
Comment (a7vicky) [reply go.mod]: Thank you Ravi for the review. I have aligned dependencies accordingly

**Comment by a7vicky** [comment]
tide/merge-method-squash

**Pull request closed without merging**